File: src/main/java/io/github/toolsjs/kubejs/builders/PaxelItemBuilder.java
package io.github.toolsjs.kubejs.builders;

import com.blakebr0.cucumber.item.tool.BasePaxelItem;
import dev.latvian.mods.kubejs.item.custom.HandheldItemBuilder;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.Item;

// tysm to https://github.com/FooterMan15/PaxelJS

public class PaxelItemBuilder extends HandheldItemBuilder {
    public PaxelItemBuilder(ResourceLocation i) {
        super(i, 0.5f, -2f);
    }

    @Override
    public Item createObject() {
        return new BasePaxelItem(toolTier,properties -> createItemProperties());
    }
}

File: src/main/kotlin/io/github/toolsjs/kubejs/builders/cucumber/PaxelItemBuilder.kt
package io.github.toolsjs.kubejs.builders.cucumber

import com.blakebr0.cucumber.item.tool.BasePaxelItem
import dev.latvian.mods.kubejs.item.custom.HandheldItemBuilder
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.Item
import java.util.function.IntSupplier

// tysm to https://github.com/FooterMan15/PaxelJS
class PaxelItemBuilder(i: ResourceLocation?) : HandheldItemBuilder(i, 0.5f, -2f) {
    override fun createObject(): Item {
        return BasePaxelItem(toolTier) { properties: Item.Properties? -> createItemProperties() }
    }
}